### PR TITLE
Sparse fix for epel 0.6.4 sparse - redefines

### PR DESCRIPTION
**Pull request by aversecat**
We should rely on sparse from epel to do automated sparse checking and not a git tag. But the 0.6.4 build currently fails on sparse/gcc redefines.

This magic Awk from Zach script processes sparse and gcc internal defines and leaves the one intact that sparse doesn't have.

It's wrapped in a simple "atomic" shell test to see if it exists and makes it in a tmp file which renames it into place so that `make -j` properly functions and doesn't cause a midway truncated file.

### CODE REVIEW

**Comment by versity-github** [comment]
Build 198 http://jenkins.vpn.versity.com:8080/job/scoutfs-run-pr-el8/198/ git: fe3183147be11bc494eab9482a4d7f6fead870b0

**Comment by versity-github** [comment]
Build 222 http://jenkins.vpn.versity.com:8080/job/scoutfs-run-pr/222/ git: fe3183147be11bc494eab9482a4d7f6fead870b0

**Comment by versity-github** [comment]
Build 200 http://jenkins.vpn.versity.com:8080/job/scoutfs-run-pr-el8/200/ git: fe3183147be11bc494eab9482a4d7f6fead870b0

**Comment by versity-github** [comment]
Build 224 http://jenkins.vpn.versity.com:8080/job/scoutfs-run-pr/224/ git: fe3183147be11bc494eab9482a4d7f6fead870b0

File: utils/sparse.sh [hunk 1]
```diff
@@ -35,12 +35,27 @@ RE="$RE|error: undefined identifier '__builtin_fpclassify'"
 search=$(gcc -print-search-dirs | awk '($1 == "install:"){print "-I" $2}')
 
 #
-# We're trying to use sparse against glibc headers which go wild trying to
```
Comment (zabbo) [inline utils/sparse.sh:38]: Don't remove the comment that describes  why this nonsense is in here.  The text you ended up with should replace the "But not __SIZE_TYPE" as that's all that you're changing, and leave the initial text.

File: utils/sparse.sh [hunk 2]
```diff
@@ -35,12 +35,27 @@ RE="$RE|error: undefined identifier '__builtin_fpclassify'"
...
-# use internal compiler macros to test features.  We copy gcc's and give
-# them to sparse.  But not __SIZE_TYPE__ 'cause sparse defines that one.
+# Filter out whatever gcc defines we already have from the sparse defines
+# so they don't throw redefined errors.
 #
 defines=".sparse.gcc-defines.h"
-gcc -dM -E -x c - < /dev/null | grep -v __SIZE_TYPE__ > $defines
+if [[ ! -f $defines ]]; then
```
Comment (zabbo) [inline utils/sparse.sh:42]: This now never recalculates the defines if the toolchain changes.  Get rid of the test and mv.  Append .$$ to the defines="" file name so each invocation uses its own file.  Then remove it after the sparse invocation.  It'll still have the prefix that'll be removed by make clean if it gets interrupted or crashes.

**Review by zabbo — CHANGES_REQUESTED** [review]

**Comment by versity-github** [comment]
Build 209 http://jenkins.vpn.versity.com:8080/job/scoutfs-run-pr-el8/209/ git: 3b8d2eab8ef7066cf0d9cd67c838ba1b03c68773

**Comment by versity-github** [comment]
Build 233 http://jenkins.vpn.versity.com:8080/job/scoutfs-run-pr/233/ git: 3b8d2eab8ef7066cf0d9cd67c838ba1b03c68773

**Pull request merged**